datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaBasedConnectorTaskMetrics.java

```diff
@@ -130,6 +143,43 @@
     DYNAMIC_METRICS_MANAGER.registerGauge(_className, AGGREGATE, NUM_TOPICS, aggNumTopics::get);
   }
 
+  /**
+   * Register some of the Kafka consumer metrics of interest. This cannot be done as part of the constructor
+   * as the Kafka consumer may be created at a later time than this object.
+   * @param consumer the Kafka consumer for which to register the metrics
+   * @param clientId the Kafka consumer's client.id
+   */
+  public void registerKafkaConsumerMetrics(Consumer<?, ?> consumer, String clientId) {
+    if (consumer == null || StringUtils.isBlank(clientId)) {
```

Comment:
Why not throw on null consumer (and possibly `clientId` as well)?

Reply:
I've removed the consumer check, as if the consumer is null, we _should_ throw. I do believe that clientId isn't a mandatory consumer config, due to which I'd like to ensure that we don't throw an exception if this is blank, so keeping this one around.

Reply:
Makes sense. Feel free to add a validate.notnull for `consumer`, too.